### PR TITLE
Remove practically unused cache interface

### PR DESCRIPTION
**Pull request by greatroar**
What is the purpose of this change? What does it change?
--------------------------------------------------------

It removes the restic.Cache interface, which was unused in the sense that exactly one code path passed values of it around, only to type-assert the values back to *cache.Cache at the point of use. It also removes an unused error type from the cache package.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

No.

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [ ] I have added tests for all changes in this PR
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: internal/restic/cache.go [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-	// Wrap returns a backend with a cache.
-	Wrap(Backend) Backend
-
-	// IsNotExist returns true if the error was caused by a non-existing file.
-	IsNotExist(err error) bool
-
-	// Load returns a reader that yields the contents of the file with the
-	// given id if it is cached. rd must be closed after use. If an error is
-	// returned, the ReadCloser is nil. The files are still encrypted
-	Load(h Handle, length int, offset int64) (io.ReadCloser, error)
```
Comment (MichaelEischer) [inline internal/restic/cache.go:19]: Please lower-case `Load`, `SaveWriter` and `Remove` in `internal/restic/file.go` to private methods. These aren't used anywhere outside the cache

**Review by MichaelEischer — COMMENTED** [review]
Sorry for taking so long to review this, I just couldn't decide whether we should keep the Cache interface or not. But as the Cache is really only ever used by the actual repository implementation, I guess there's really not much use in keeping it.

Please rebase this PR to the current master.

File: internal/restic/cache.go (hunk 1 above)
Comment (greatroar) [reply internal/restic/cache.go:19]: Done.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**